Objective-C/CBLReplicator.mm

```diff
@@ -808,7 +803,9 @@ - (bool) filterDocument: (C4CollectionSpec)c4spec
             return NO;
         }
     } else {
-        collection = [_config.database defaultCollectionOrThrow];
+        collection = [_collectionMap objectForKey: $sprintf(@"%@.%@",
```

Comment:
Same as above, no need to have if/else because c4spec.name.buf will always have value unless there is a bug.

```
NSString* name = slice2string(c4spec.name);
NSString* scopeName = slice2string(c4spec.scope);
CBLCollection collection = [_collectionMap objectForKey: $sprintf(@"%@.%@", scopeName, name)];
Assert(collection, "Collection is not found in the replicator config when calling the filter function.");
```